 /**
  * The persisted queued.
  * The entry status is not saved, it's defined during the download process.
  * Version : 4
  * All string are encoded in UTF-8.
  */

syntax = "proto3";

import "common.proto";

package Protos.Queue;

message Queue {
   message Entry {
      // Values must compatible with 'GUI::State::Download::Status'.
      enum Status {
         QUEUED = 0x0;
         COMPLETE = 0x3;
         PAUSED = 0x4;
      }
      Common.Entry remote_entry = 1;
      Common.Entry local_entry = 2;

      Common.Hash peer_source_id = 3;
      string peer_source_nick = 4;

      Status status = 5; // [default = QUEUED] Only valid for Common.Entry.type == FILE.
   }

   uint32 version = 1;
   repeated Entry entry = 2;
}